service/controller/resource/promxy/create.go

```diff
@@ -0,0 +1,53 @@
+package promxy
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/prometheus-meta-operator/service/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "checking if promxy config map already exists")
+	configMap, err := r.k8sClient.K8sClient().CoreV1().ConfigMaps(key.PromxyConfigMapNamespace()).Get(ctx, key.PromxyConfigMapName(), metav1.GetOptions{})
+
+	if apierrors.IsNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "promxy needs to be updated")
+		return nil
```

Comment:
This is not where the decision is being made to update or not the configmap. And why do we return nil ?
I think we should return any error which occurs here: when we do not find the configmap there's a bigger issue and this resource cannot do its job.

Reply:
Should I still log that promxy needs to be updated?